frontend/packages/pipelines-plugin/src/components/pipelines/detail-page-tabs/pipeline-details/PipelineDetails.tsx

```diff
@@ -20,13 +20,9 @@ const PipelineDetails: React.FC<PipelineDetailsProps> = ({
   customData: routeTemplates,
 }) => {
   const { t } = useTranslation();
-  const taskLinks = pipeline.spec.tasks
-    .filter((pipelineTask: PipelineTask) => !!pipelineTask.taskRef)
-    .map((task) => ({
-      model: getResourceModelFromTaskKind(task.taskRef.kind),
-      name: task.taskRef.name,
-      displayName: task.name,
-    }));
+  const taskLinks = getPipelineTaskLinks(pipeline);
+  const finallyTaskLinks = getPipelineTaskLinks(pipeline, true);
```

Comment:
Instead of this i would suggest changing the `getPipelineTaskLinks` function such that it returns both `tasks` and `finallyTasks`:
eg:
```
 const { tasksLinks ,  finallyTaskLinks } = getPipelineTaskLinks(pipeline);
```